### ISSUE TITLE
Edited RAW files exported as TIFF not JPEG

### ISSUE THREAD

**Issue by neilpa** [RhetTbull/osxphotos#374]
Looks like edited RAW photos (`NEF` in my case) are exported as `TIFF` and not `JPEG`. This was without any explicit use of `--convert-to-jpeg` or `--use-photokit`.

```
Exporting NIK_7078.NEF (61777198-F54C-4041-BABB-79318E615526.nef) as 08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526.JPEG
Writing XMP sidecar /Volumes/sandisk-hfsp/pics/backup/2020/08/09/08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526.JPEG.xmp
Writing metadata with exiftool for /Volumes/sandisk-hfsp/pics/backup/2020/08/09/08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526.JPEG
exiftool error for file /Volumes/sandisk-hfsp/pics/backup/2020/08/09/08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526.JPEG: Error: Not a valid JPEG (looks more like a TIFF) - /Volumes/sandisk-hfsp/pics/backup/2020/08/09/08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526.JPEG
Error exporting photo (61777198-F54C-4041-BABB-79318E615526: NIK_7078.NEF) as /Volumes/sandisk-hfsp/pics/backup/2020/08/09/08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526.JPEG: Error: Not a valid JPEG (looks more like a TIFF) - /Volumes/sandisk-hfsp/pics/backup/2020/08/09/08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526.JPEG
```

All were edited with the builtin Photos editor, mostly just crops. Here's the version and command I was using

```
osxphotos --version
osxphotos, version 0.40.8

osxphotos export \
  --db $DIR/macos.photoslibrary \
  --directory '{created.year}/{created.mm}/{created.dd}' \
  --filename '{created.hour}-{created.min}-{created.sec}.{original_name}.{uuid}' \
  --exiftool \
  --sidecar xmp \
  --jpeg-ext JPEG \
  --strip \
  --verbose \
  --report $DIR/backup/report.$TS.csv \
  $DIR/backup
```

Double checked the number of `TIFF` errors in the CSV report against a smart album with `is RAW && is EDITED` and the counts matched. (There were some other unrelated errors that I haven't looked at yet).

**Comment by RhetTbull** [comment]
From the data you posted, this looks like an osxphotos bug—it’s changing the extension when it shouldn’t.  I’ll see if I can replicate this.  What version of MacOS are you using?

**Comment by neilpa** [comment]
This is on Catalina. I experimented a bit more and the problem appears to be with `--jpeg-ext JPEG`. Removing that fixed the export.

```
Exporting 1 photo to /Volumes/sandisk-hfsp/pics/backup...
Exporting NIK_7078.NEF (61777198-F54C-4041-BABB-79318E615526.nef) as 08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526.NEF
Writing XMP sidecar /Volumes/sandisk-hfsp/pics/backup/2020/08/09/08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526.NEF.xmp
Writing metadata with exiftool for /Volumes/sandisk-hfsp/pics/backup/2020/08/09/08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526.NEF
Exporting edited version of NIK_7078.NEF (61777198-F54C-4041-BABB-79318E615526.nef) as 08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526_edited.jpeg
Writing XMP sidecar /Volumes/sandisk-hfsp/pics/backup/2020/08/09/08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526_edited.jpeg.xmp
Writing metadata with exiftool for /Volumes/sandisk-hfsp/pics/backup/2020/08/09/08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526_edited.jpeg
Exported /Volumes/sandisk-hfsp/pics/backup/2020/08/09/08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526.NEF
Exported /Volumes/sandisk-hfsp/pics/backup/2020/08/09/08-17-59.NIK_7078.61777198-F54C-4041-BABB-79318E615526_edited.jpeg
Writing export report to /Volumes/sandisk-hfsp/pics/backup/report.2021-02-09_17-02-38.csv
Processed: 1 photo, exported: 2, missing: 0, error: 0
```

**Comment by RhetTbull** [comment]
Thanks.  I’ll look into it.  That shouldn’t change anything if the file isn’t a jpeg so clearly a bug.

**Comment by RhetTbull** [comment]
I *think* I've fixed this in [v0.40.13](https://github.com/RhetTbull/osxphotos/releases/tag/v0.40.13).  The code was checking the current UTI of the file instead of the UTI of the original file when exporting originals so files that had been edited appeared to have the wrong UTI when exporting the original.  The code for --jpeg-ext has proven to be surprisingly difficult to get right--let me know if you still experience problems with it.

**Comment by neilpa** [comment]
That did the trick. Exported my full library using that option without any errors.
